src/Numerics/Mesh/BrickMesh.jl

```diff
@@ -1126,4 +1126,341 @@ function enumerateboundaryfaces!(elemtoelem, elemtobndy, periodicity, boundary)
     return (bndytoelem, bndytoface)
 end
 
+"""
+    connectmeshfull(comm::MPI.Comm, elemtovert, elemtocoord, elemtobndy,
+                faceconnections)
+
+This function takes in a mesh (as returned for example by `brickmesh`) and
+returns a corner connected mesh.  This returns a `NamedTuple` of:
+
+ - `elems` the range of element indices
+ - `realelems` the range of real (aka nonghost) element indices
+ - `ghostelems` the range of ghost element indices
+ - `ghostfaces` ghost element to face is received;
+   `ghostfaces[f,ge] == true` if face `f` of ghost element `ge` is received.
+ - `sendelems` an array of send element indices
+ - `sendfaces` send element to face is sent;
+   `sendfaces[f,se] == true` if face `f` of send element `se` is sent.
+ - `elemtocoord` element to vertex coordinates; `elemtocoord[d,i,e]` is the
+    `d`th coordinate of corner `i` of element `e`
+ - `elemtoelem` element to neighboring element; `elemtoelem[f,e]` is the
+   number of the element neighboring element `e` across face `f`.  If there is
+   no neighboring element then `elemtoelem[f,e] == e`.
+ - `elemtoface` element to neighboring element face; `elemtoface[f,e]` is the
+   face number of the element neighboring element `e` across face `f`.  If
+   there is no neighboring element then `elemtoface[f,e] == f`.
+ - `elemtoordr` element to neighboring element order; `elemtoordr[f,e]` is the
+   ordering number of the element neighboring element `e` across face `f`.  If
+   there is no neighboring element then `elemtoordr[f,e] == 1`.
+ - `elemtobndy` element to bounday number; `elemtobndy[f,e]` is the
+   boundary number of face `f` of element `e`.  If there is a neighboring
+   element then `elemtobndy[f,e] == 0`.
+ - `nabrtorank` a list of the MPI ranks for the neighboring processes
+ - `nabrtorecv` a range in ghost elements to receive for each neighbor
+ - `nabrtosend` a range in `sendelems` to send for each neighbor
+
+The algorithm assumes that the 2-D mesh can be gathered on single process,
```

Comment:
Is this 2D in the sense that it is a grid made of horizontal layers?

Reply:
This is done at the 2D mesh stage. This 2D mesh is eventually extruded in the third direction, by stacking, to generate the 3D outer mesh.